JumpscaleCore/core/BASECLASSES/JSConfigsBCDB.py

```diff
@@ -281,19 +282,13 @@ def delete(self, name=None):
         self._delete(name=name)
 
     def _delete(self, name=None):
-        self._model
         if name:
-            res = self._findData(name=name)
-            if len(res) == 0:
-                return
-            elif len(res) == 1:
-                self._model.delete(res[0].id)
+            _, child = self._get(name=name)
```

Comment:
`_get` will get it from self._children if it's there, otherwise from the database. This way we avoid having to do it over two steps.